elide-async/src/test/java/com/yahoo/elide/async/service/AsyncExecutorServiceTest.java

```diff
@@ -72,25 +66,33 @@ public void testAsyncExecutorServiceSet() {
         assertNotNull(service.getRunners());
         assertEquals(60, service.getMaxRunTime());
         assertNotNull(service.getExecutor());
-        assertNotNull(service.getInterruptor());
+        assertNotNull(service.getUpdater());
         assertEquals(asyncQueryDao, service.getAsyncQueryDao());
     }
 
+    //Test for regular query execution before asyncAfterSeconds expires
     @Test
-    public void testExecuteQuery() {
+    public void testExecuteQuery() throws ExecutionException, TimeoutException, InterruptedException {
 
-        service.executeQuery(queryObj, testUser, NO_VERSION);
-        verify(asyncQueryDao, times(0)).updateStatus(queryObj.getId(), QueryStatus.QUEUED);
+       when(queryObj.getAsyncAfterSeconds()).thenReturn(10);
+       service.executeQuery(queryObj, testUser, NO_VERSION);
+       verify(queryObj, times(1)).setStatus(QueryStatus.PROCESSING);
+       verify(queryObj, times(1)).setStatus(QueryStatus.COMPLETE);
 
     }
 
+
+    //Test for TimeoutException after asyncAfterSeconds
     @Test
-    public void testFutureTimeoutException() throws InterruptedException, ExecutionException, TimeoutException {
-         when(queryObj.getAsyncAfterSeconds()).thenReturn(1);
-         service.executeQuery(queryObj, testUser, NO_VERSION);
-         Future<?> task = mock(Future.class);
-         when(task.get(anyLong(), any(TimeUnit.class))).thenThrow(TimeoutException.class);
-         verify(asyncQueryDao, times(0)).updateStatus(queryObj.getId(), QueryStatus.QUEUED);
+    public void testAsyncAfterExecuteQuery() throws InterruptedException {
+
+        AsyncQuery query = new AsyncQuery();
+        query.setAsyncAfterSeconds(0);
+        String id = "edc4a871-dff2-4054-804e-d80075cf827d";
+        query.setId(id);
+        User user = new User(null);
+        service.executeQuery(query, user, NO_VERSION);
+        verify(asyncQueryDao, times(1)).updateAsyncQueryResult(queryResultObj, query.getId());
```

Comment:
Should we also verify 
       verify(queryObj, times(1)).setStatus(QueryStatus.PROCESSING);
      verify(queryObj, times(1)).setStatus(QueryStatus.COMPLETE);